### PR TITLE
PAYARA-3859: Update Microprofile Rest Client

### PR DESCRIPTION
**Pull request by pdudits**
Updates REST Client to version from payara/Payara_PatchedProjects#248

The test case is present in pdudits/data-replication-talk (`ReplicationIT` in `consumer-app`).

Another testcase was added to [upstream](https://github.com/payara/patched-src-jersey/pull/20/files#diff-2e366f66b6244208226da5cbdbc5ba63)

### CODE REVIEW

File: appserver/packager/microprofile-package/pom.xml [hunk 1]
```diff
@@ -224,7 +224,7 @@
             <artifactId>jersey-proxy-client</artifactId>
         </dependency>
         <dependency>
-            <groupId>org.glassfish.jersey.ext</groupId>
```
Comment (MarkWareham) [inline appserver/packager/microprofile-package/pom.xml]:
```
[ERROR]   The project org.glassfish.main.packager:microprofile-package:5.192-SNAPSHOT (/home/mark/git/payara/appserver/packager/microprofile-package/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.version' for org.glassfish.jersey.ext.microprofile:jersey-mp-rest-client:jar is missing. @ org.glassfish.main.packager:microprofile-package:[unknown-version], /home/mark/git/payara/appserver/packager/microprofile-package/pom.xml, line 226, column 21
```
without looking, due to dependency managment needing updating too perhaps?

**Comment by pdudits** [comment]
jenkins test please

File: appserver/packager/microprofile-package/pom.xml (hunk 1 above)
Comment (pdudits) [reply appserver/packager/microprofile-package/pom.xml]: BOM changed in payara/Payara_PatchedProjects#248. After clearing local repo for `org.glassfish.jersey:jersey-bom` it should build fine.

**Comment by pdudits** [comment]
jenkins test please

**Comment by pdudits** [comment]
jenkins test please

**Comment by Cousjava** [comment]
Jenkins test please

**Pull request merged**